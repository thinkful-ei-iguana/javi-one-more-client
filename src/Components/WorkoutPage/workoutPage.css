a{
    text-decoration: none;
}



.workout-title{
    display: inline-block;
    border: 1px solid grey;
    padding: 10px;
    margin-top: 5px;
}
@media (min-width: 320px){
    .your-workouts-title{
        font-size: 1rem;
    }
    
    .add-workout{
        font-size: 12px;
    }

    .workout-title{
        font-size: 1rem;
    }
}

@media (min-width: 320px){
    .your-workouts-title{
        font-size: 1rem;
    }
    
    .add-workout{
        font-size: 12px;
    }

    .workout-title{
        font-size: 1rem;
    }
}

@media (min-width: 375px){
    .your-workouts-title{
        font-size: 1.2rem;
    }
    
    .add-workout{
        font-size: 12px;
    }

    .workout-title{
        font-size: 1.2rem;
    }
}
@media (min-width: 425px){
    .your-workouts-title{
        font-size: 1.2rem;
    }
    
    .add-workout{
        font-size: 14px;
    }

    .workout-title{
        font-size: 1.2rem;
    }
}

@media (min-width: 768px){
    .your-workouts-title{
        font-size: 1.5rem;
    }
    
    .add-workout{
        font-size: 1rem;
    }

    .workout-title{
        font-size: 1.5rem;
    }
}