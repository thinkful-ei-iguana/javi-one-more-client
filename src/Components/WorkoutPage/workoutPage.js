import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import Header from '../Header/Header.js'
import '../WorkoutPage/workoutPage.css'
import WorkoutList from '../WorkoutList/WorkoutList.js'
import config from '../../config';


class WorkoutPage extends Component {
  state = {
      workouts: [],
      error: null,
  }

  setWorkouts = workouts => {
      this.setState({
          workouts,
          error: null,
      })
    //   console.log(this.state.workouts)
  }



componentDidMount(){
    fetch(config.API_ENDPOINT, {
        method: 'GET',
        headers: {
            'content-type': 'application/json'
        }
    })
    .then(res => {
        if(!res.ok){
            throw new Error(res.status)
        }
        return res.json()
    })
    .then(this.setWorkouts)
    .catch(error => this.setState({error}))
}

renderWorkoutInfo(){
    
}


    render() {
        
        return (
            <div>
                <Header />
                <main>
                <Link to="/addWorkout"><span className="add-workout">Add Workout</span></Link>
                <WorkoutList history={this.props} workouts={this.state.workouts}/>
                </main>
            </div>
        )
    }
}

export default WorkoutPage
