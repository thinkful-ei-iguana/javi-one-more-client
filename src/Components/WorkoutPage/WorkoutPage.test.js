import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import WorkoutPage from '../WorkoutPage/WorkoutPage';
import renderer from 'react-test-renderer'


describe('add workout' , () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><WorkoutPage /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><WorkoutPage /></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot();
  })
})