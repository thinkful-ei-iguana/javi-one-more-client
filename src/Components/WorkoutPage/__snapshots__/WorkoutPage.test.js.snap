// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add workout renders the UI as expected 1`] = `
<div
  className="workoutPage-container"
>
  <div
    className="header-container"
  >
    <header
      className="App-header"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <span>
          One More
        </span>
      </a>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-heartbeat fa-w-16 "
        data-icon="heartbeat"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M320.2 243.8l-49.7 99.4c-6 12.1-23.4 11.7-28.9-.6l-56.9-126.3-30 71.7H60.6l182.5 186.5c7.1 7.3 18.6 7.3 25.7 0L451.4 288H342.3l-22.1-44.2zM473.7 73.9l-2.4-2.5c-51.5-52.6-135.8-52.6-187.4 0L256 100l-27.9-28.5c-51.5-52.7-135.9-52.7-187.4 0l-2.4 2.4C-10.4 123.7-12.5 203 31 256h102.4l35.9-86.2c5.4-12.9 23.6-13.2 29.4-.4l58.2 129.3 49-97.9c5.9-11.8 22.7-11.8 28.6 0l27.6 55.2H481c43.5-53 41.4-132.3-7.3-182.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <a
        href="/workoutPage"
        onClick={[Function]}
      >
        <span
          className="enter-app"
        >
          Enter App
        </span>
      </a>
    </header>
  </div>
  <main
    id="background-image"
  >
    <a
      href="/addWorkout"
      onClick={[Function]}
    >
      <span
        className="add-workout"
      >
        + Workout
      </span>
    </a>
    <div
      className="your-workouts-container"
    >
      <h2
        className="your-workouts-title"
      >
        Your Workouts
      </h2>
    </div>
  </main>
  <div>
    <footer>
      copyright 2019, one more
    </footer>
  </div>
</div>
`;
